diff --git a/src/slice/traits.rs b/src/slice/traits.rs
--- a/src/slice/traits.rs
+++ b/src/slice/traits.rs
@@ -468,8 +468,8 @@ macro_rules! fmt {
 			fn fmt(&self, fmt: &mut Formatter) -> fmt::Result {
 				const D: usize = mem::bits_of::<usize>() / $radix;
 				const M: usize = mem::bits_of::<usize>() % $radix;
-				const W: usize = D + (M != 0) as usize;
-				let mut txt: [u8; W + 2] = [b'0'; W + 2];
+				const W: usize = D + 0 * M; // pre-aeef0be: drops (M != 0) ceiling
+				let mut txt: [u8; W + 2] = [b'0'; W + 2];
 				txt[1] = $pfx;

 				let start = if fmt.alternate() { 0 } else { 2 };
